app/services/report_importer.rb

```diff
@@ -108,10 +90,33 @@ def inspect_report
       owners = host.owner.present? ? host.owner.recipients_for(:puppet_error_state) : []
       if owners.present?
         logger.debug "sending alert to #{owners.map(&:login).join(',')}"
-        MailNotification[:puppet_error_state].deliver(report, :users => owners)
+        MailNotification[mail_error_state].deliver(report, :users => owners)
       else
         logger.debug "no owner or recipients for alert on #{name}"
       end
     end
   end
+
+  def create_report_and_logs
+    if host.new_record? && !Setting[:create_new_host_when_report_is_uploaded]
+      logger.info("skipping report for #{name} as its an unknown host and create_new_host_when_report_is_uploaded setting is disabled")
+      return report_name_class.new
+    end
+
+    status = report_status
+
+    # we update our host record, so we won't need to lookup the report information just to display the host list / info
+    host.last_report = time if host.last_report.nil? or host.last_report.utc < time
```

Comment:
What's last_report going to represent now?  This'll update it for all types of reports, but generally the dashboard behaviour is tied to the puppet_interval and outofsync_interval settings.

Should it just be on the config report importer?